src/components/gitProvider/GitProvider.tsx

```diff
@@ -1,41 +1,135 @@
-import React, { useState } from 'react';
-import { useLocation, useHistory, useRouteMatch } from 'react-router'
-import { getGitProviderList, saveGitProviderConfig, updateGitProviderConfig } from './service'
-import { showError, useForm, useEffectAfterMount, useAsync, Progressing } from '../common'
+import React, { useState, useEffect } from 'react';
+import { getGitHostList, getGitProviderList } from '../../services/service';
+import { saveGitHost, saveGitProviderConfig, updateGitProviderConfig } from './gitProvider.service'
+import { showError, useForm, useEffectAfterMount, useAsync, Progressing, ErrorScreenManager } from '../common'
 import { List, CustomInput, ProtectedInput } from '../globalConfigurations/GlobalConfiguration'
 import { toast } from 'react-toastify'
-import Tippy from '@tippyjs/react';
 import { DOCUMENTATION } from '../../config';
-import { GlobalConfigCheckList } from '../checkList/GlobalConfigCheckList';
 import { ReactComponent as GitLab } from '../../assets/icons/git/gitlab.svg'
 import { ReactComponent as Git } from '../../assets/icons/git/git.svg'
 import { ReactComponent as GitHub } from '../../assets/icons/git/github.svg'
 import { ReactComponent as BitBucket } from '../../assets/icons/git/bitbucket.svg'
+import { styles } from './gitProvider.util';
+import Tippy from '@tippyjs/react';
+import CreatableSelect from 'react-select/creatable';
+interface Githost {
+    id: number;
+    name: string;
+    active: boolean;
+    eventTypeHeader: string;
+    secretHeader: string;
+    webhookSecret: string;
+    webhookUrl: string;
+}
 
 export default function GitProvider({ ...props }) {
-    const location = useLocation();
-    const match = useRouteMatch();
-    const history = useHistory();
     const [loading, result, error, reload] = useAsync(getGitProviderList)
-    if (loading && !result) return <Progressing pageLoader />
-    if (error) {
-        showError(error)
-        if (!result) return null
+    const [providerList, setProviderList] = useState([])
+    const [hostList, setHostList] = useState([])
+    const [hostListOption, setHostListOption] = useState([])
+    const [isPageLoading, setIsPageLoading] = useState(true)
+    const [isErrorLoading, setIsErrorLoading] = useState(false)
+    const [errors, setErrors] = useState([])
+    async function getInitData() {
+        try {
+            const { result: providers = [] } = await getGitProviderList()
+            const { result: hosts = [] } = await getGitHostList()
```

Comment:
use [result1, result2] = Promise.all([async1(), async2()]); for line no 35 and 36